p2p_wallet/Scenes/Main/Send/SendСoordinator.swift

```diff
@@ -203,31 +203,36 @@ class SendCoordinator: Coordinator<SendResult> {
         
         coordinate(to: SendInputCoordinator(
             recipient: recipient,
-            preChosenWallet: nil,
-            preChosenAmount: nil,
+            preChosenWallet: preChosenWallet,
+            preChosenAmount: preChosenAmount,
             navigationController: rootViewController,
             source: .none,
             allowSwitchingMainAmountType: true,
             sendViaLinkSeed: seed
         ))
-            .sink { [weak self] result in
-                switch result {
-                case let .sent(transaction):
-                    self?.result.send(.sent(transaction))
-                case let .sentViaLink(link, transaction):
-                    self?.startSendViaLinkCompletionFlow(
-                        link: link,
-                        formatedAmount: transaction.amount.tokenAmountFormattedString(symbol: transaction.walletToken.token.symbol),
-                        transaction: transaction
-                    )
-                case .cancelled:
-                    break
-                }
+        .sink { [weak self] result in
+            switch result {
+            case let .sent(transaction):
+                self?.result.send(.sent(transaction))
+            case let .sentViaLink(link, transaction):
+                self?.startSendViaLinkCompletionFlow(
+                    link: link,
+                    formatedAmount: transaction.amount.tokenAmountFormattedString(symbol: transaction.walletToken.token.symbol),
+                    transaction: transaction
+                )
+            case .cancelled:
+                break
             }
-            .store(in: &subscriptions)
+        }
+        .store(in: &subscriptions)
     }
     
-    private func startSendViaLinkCompletionFlow(link: String, formatedAmount: String, transaction: SendTransaction) {
+    private func startSendViaLinkCompletionFlow(
+        link: String,
+        formatedAmount: String,
+        transaction: SendTransaction
+    ) {
+        var withError = false
```

Comment:
Let's change the `SendCreateLinkCoordinator`'s result to Bool or something, so we 
can return `withError` from the result directly, and we don't have to use `withError` local variable